Services/Mail/classes/Setup/class.ilMailDatabaseUpdateSteps.php

```diff
@@ -31,4 +31,54 @@ public function step_1() : void
             $this->db->dropTableColumn('mail', 'm_email');
         }
     }
+
+    public function step_2() : void
+    {
+        $result = $this->db->queryF('SELECT value FROM settings WHERE module = %s AND keyword = %s', ['text', 'text'], ['common', 'mail_system_sys_signature']);
+        $row = $this->db->fetchAssoc($result);
+        if (isset($row['value']) && $row['value'] !== null) {
+            $new_value = str_replace(['[CLIENT_NANE]', '[CLIENT_DESC]', '[CLIENT_URL]'], ['[INSTALLATION_NAME]', '[INSTALLATION_DESC]', '[ILIAS_URL]'], $row['value']);
+            if ($new_value !== $row['value']) {
+                $this->db->manipulateF(
+                    'UPDATE settings SET value = %s WHERE module = %s AND keyword = %s',
+                    ['text', 'text', 'text'],
+                    [$new_value, 'common', 'mail_system_sys_signature']
+                );
+            }
+        }
+    }
+
+    public function step_3() : void
+    {
+        $result = $this->db->query("SELECT tpl_id, m_message FROM mail_man_tpl WHERE m_message LIKE '%[CLIENT_NAME]%'");
+        while ($row = $this->db->fetchAssoc($result)) {
+            if (isset($row['m_message'], $row['tpl_id']) && $row['m_message'] !== null && $row['tpl_id'] !== null) {
+                $new_value = str_replace('[CLIENT_NANE]', '[INSTALLATION_NAME]', $row['m_message']);
+                if ($new_value !== $row['m_message']) {
+                    $this->db->manipulateF(
+                        'UPDATE mail_man_tpl SET m_message = %s WHERE tpl_id = %s',
+                        ['text', 'text'],
+                        [$new_value, $row['tpl_id']]
+                    );
+                }
+            }
+        }
+    }
+
+    public function step_4() : void
+    {
+        $result = $this->db->query("SELECT lang, type, body FROM mail_template WHERE body LIKE '%[CLIENT_NAME]%'");
+        while ($row = $this->db->fetchAssoc($result)) {
+            if (isset($row['lang'], $row['type'], $row['body']) && $row['body'] !== null) {
```

Comment:
```suggestion
            if (isset($row['lang'], $row['type'], $row['body'])) {
```

If `isset($row['lang'], $row['type'], $row['body'])` evaluates to `true`, the values cannot be `null`, so the further check is obsolete.